src/components/NavScrollItem.js

```diff
@@ -37,79 +35,64 @@ const defaultProps = {
   element: "li",
 }
 
-// A Scroll Aware NavItem compatible with React Bootstrap's Nav
-class NavScrollItem extends React.Component {
-  handleClick = (e) => {
-    const { eventKey, onSelect, onClick, href, history, to } = this.props
+// A Scroll Aware NavItem compatible
+const NavScrollItem = ({
+  active,
+  className,
+  style,
+  activeClass,
+  href,
+  hashSpy,
+  onClick,
+  to,
+  spy,
+  smooth,
+  duration,
+  children,
+  element: Element,
+}) => {
+  const location = useLocation()
+  const root = process.env.GATSBY_BRANCH
+    ? `/${process.env.GATSBY_BRANCH}/`
+    : "/"
 
+  const handleClick = (e) => {
     if (typeof onClick === "function") {
       onClick()
     }
 
     if (href) {
       e.stopPropagation()
 
-      history.push(`${href}#${to || ""}`)
-    }
-
-    if (onSelect) {
-      e.preventDefault()
-
-      onSelect(eventKey, e)
+      window.history.pushState({}, "", `${href}#${to || ""}`)
```

Comment:
Out of curiosity, where was `history` coming from before?

Reply:
Bananas, I lied here---it doesn't look like the `pushState` is taking into account the current branch 🤔  e.g. going to https://preview.keep.network/gatsby-fix-nav/ and then clicking on `Team` will update the URL to `https://preview.keep.network/#team` instead of `https://preview.keep.network/gatsby-fix-nav/#team`.

Reply:
Needs a bit of fixing up...will circle back in a later PR! Before history was coming from the react router redux store.